### PR TITLE
WIP: add page type identification module

### PR DESCRIPTION
**Pull request by ishank-dev**
Need help in fetching the xml file directly using OL python library.
Everything else is still a work in progress

### CODE REVIEW

File: bgp/modules/terms.py [hunk 1]
```diff
@@ -3,6 +3,8 @@
 import string
 import re
 from collections import defaultdict
+from lxml import etree
```
Comment (tabshaikh) [inline bgp/modules/terms.py:6]: Could you add lxml==4.5.0 requirement to requirements.txt

File: bgp/modules/terms.py [hunk 2]
```diff
@@ -113,3 +115,37 @@ def validate_isbn(isbn):
...
+    def __init__(self, modules):
+        """
+        :param modules: a dict of {'name': module}
+        """
+        self.modules = modules
+
+class CopyrightPageDetectorModule:
+    
+    def __init__(self, keyword):
+        self.keyword = keyword
```
Comment (tabshaikh) [inline bgp/modules/terms.py:130]: Why would the CopyrightPageDetectorModule need the parameter keyword shouldn't the keyword `Copyright`. Shouldnt the self.keyword be assigned to `copyright`?

File: bgp/modules/terms.py [hunk 3]
```diff
@@ -113,3 +115,37 @@ def validate_isbn(isbn):
...
+
+class PageTypeProcessor():
+
+    def __init__(self, modules):
+        """
+        :param modules: a dict of {'name': module}
+        """
+        self.modules = modules
+
+class CopyrightPageDetectorModule:
```
Comment (tabshaikh) [inline bgp/modules/terms.py:127]: The CopyrightPageDetectorModule should inherit the base class which in this case would be PageTypeProcessor.

**Comment by ishank-dev** [comment]
I have added a fresh PR for integration so closing this now.

**Pull request closed without merging**